### DISCUSSION TITLE
Does superset support clickhouse with  certificate

### DISCUSSION THREAD

**Discussion by bigdata-spec** [apache/superset#26311] [Q&A / Help]
Hi，
   I want to connect clickhouse db with superset，
   when I set  **clickhouse://user:password@clickhouse.xxx.prod:443/default?protocol=https&ssl=true&sslmode=none&sslrootcert=/app/resources/bd_domain.crt**
or  **clickhousedb://user:password@clickhouse.xxx.prod:443/default?protocol=https&ssl=true&sslmode=none&sslrootcert=/app/resources/bd_domain.crt**

but error
![image](https://github.com/apache/superset/assets/23710717/412bd4f1-7fba-41e1-857e-61b36a9a977f)

So I want to kown  does superset support clickhouse with  certificate

**Comment by sfirke** [comment]
Looks similar to https://github.com/apache/superset/issues/14457- but I'm not sure the associated blog post really answers this.

The aspect of fetching data from your data source is handled by SQLAlchemy.  I would suggest experimenting with the SQLAlchemy connection string in a Python session, using different clickhouse-related packages (I see at least two, with the recommended one for Superset having changed recently).  If you can execute a data retrieval query from SQLAlchemy, you should be able to use that connection string in Superset.

I'm going to migrate this to a discussion, as it is not a bug report.

**Comment by clayheaton** [comment]
@bigdata-spec Did you ever figure out how to get this to work? I've been struggling with it for a few days.
